src/main/java/fi/metatavu/dcfb/server/persistence/dao/LocationDAO.java

```diff
@@ -0,0 +1,175 @@
+package fi.metatavu.dcfb.server.persistence.dao;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import fi.metatavu.dcfb.server.persistence.model.Location;
+import fi.metatavu.dcfb.server.persistence.model.Location_;
+import fi.metatavu.dcfb.server.persistence.model.LocalizedEntry;
+
+/**
+ * DAO for Location entity
+ * 
+ * @author Antti Leppä
+ */
+public class LocationDAO extends AbstractDAO<Location> {
+  
+  /**
+  * Creates new location
+  *
+  * @param id id
+  * @param slug slug
+  * @param additionalInformations additionalInformations
+  * @param country country
+  * @param latitude latitude
+  * @param longitude longitude
+  * @param name name
+  * @param postalCode postalCode
+  * @param postOffice postOffice
+  * @param streetAddress streetAddress
+  * @param lastModifier modifier
+  * @return created location
+  */
+  public Location create(UUID id, String slug, LocalizedEntry additionalInformations, String country, BigDecimal latitude, BigDecimal longitude, LocalizedEntry name, String postalCode, String postOffice, String streetAddress, UUID lastModifier) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Method has 11 parameters, which is greater than 7 authorized. ([squid:S00107](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS00107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.dcfb-api%3Adcfb-api&pullRequest=11&issues=AWQt_bglYwZU9vvrs3Km&open=AWQt_bglYwZU9vvrs3Km)

[ISSUE_KEY]: AWQt_bglYwZU9vvrs3Km